serial.c

```diff
@@ -61,15 +61,27 @@ struct sp_port *init_serial() {
   sp_free_port_list(port_list);
 
   if (m8_port != NULL) {
-    // Open the serial port and configure it
-    SDL_Log("Opening port.\n");
-    check(sp_open(m8_port, SP_MODE_READ_WRITE));
-
-    check(sp_set_baudrate(m8_port, 115200));
-    check(sp_set_bits(m8_port, 8));
-    check(sp_set_parity(m8_port, SP_PARITY_NONE));
-    check(sp_set_stopbits(m8_port, 1));
-    check(sp_set_flowcontrol(m8_port, SP_FLOWCONTROL_NONE));
+      // Open the serial port and configure it
+      SDL_Log("Opening port.\n");
+      enum sp_return result;
+
+      result = sp_open(m8_port, SP_MODE_READ_WRITE);
+      if (check(result) != SP_OK) return NULL;
```

Comment:
Since main.c is expecting a NULL return on error, I changed this to return NULL instead of exiting. This allows the program to cleanly close, flushing all buffers and closing all handles.